### PR TITLE
[Currency] Base currency's enabled status and exchange rate should not be editable

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q                    | A
| -------------       | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| related PRs | #3814 #3887 
| License       | MIT

### CODE REVIEW

File: src/Sylius/Component/Currency/Model/Currency.php [hunk 1]
```diff
@@ -111,7 +111,10 @@ public function getExchangeRate()
      */
     public function setExchangeRate($rate)
     {
-        $this->exchangeRate = $rate;
+        if(!$this->isBase())
+        {
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Currency/Model/Currency.php]: CS

File: src/Sylius/Component/Currency/Model/Currency.php [hunk 2]
```diff
@@ -175,7 +178,9 @@ public function enable()
      */
     public function disable()
     {
-        $this->enabled = false;
+        if(!$this->isBase()) {
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Currency/Model/Currency.php]: CS

**Comment by pjedrzejewski** [comment]
I think it should throw an exception. Otherwise it is a bit hidden behavior.

**Comment by michalmarcinkowski** [comment]
Definitely it should throw an exception. IMO `\LogicException` will be appropriate.

File: src/Sylius/Component/Currency/Model/Currency.php (hunk 1 above)
Comment (pamil) [inline src/Sylius/Component/Currency/Model/Currency.php]:
```php
if($this->isBase()) {
  // throw exception / return;
}

$this-exchangeRate = $rate;
```

This way the "happy" path goes at the base indentation level :)

**Comment by TheMadeleine** [comment]
:+1:  :)

**Comment by pjedrzejewski** [comment]
@TheMadeleine Don't you think it would be useful to add a message? :P

**Comment by pjedrzejewski** [comment]
Thanks Magda!

**Pull request merged**